### PR TITLE
[UPGRADE] Adds version to docset name

### PR DESCRIPTION
**Pull request by whytry313**
Very basic "add version to docset" changes, allows to generate docsets for godot 3.x and 4.x using the `version.py` file on the godot source root folder and results in a declaration using the *Godot-X.x" syntax + folder of the same name

Changes, optimizations and error handling not implemented but would be very welcomed

### CODE REVIEW

File: godotdocset.py [hunk 1]
```diff
@@ -131,23 +131,39 @@ def get_plist(name: str):
...
+	for line in open(VERSION_PATH):
+		if line.startswith("docs ="):
+			version = "-"+line.split("=")[1].strip().strip('\"')
+	return version
+
 class DocsetMaker:
 	outname = "Godot"
-	rootdir = outname + '.docset'
+	v_outname = outname+get_version()
+	# version_path = os.path.abspath(os.path.join("../", frompath, "./version.py"))
```
Comment (Yanpas) [inline godotdocset.py]: remove comment?

File: godotdocset.py [hunk 2]
```diff
@@ -131,23 +131,39 @@ def get_plist(name: str):
...
 
+def get_args_from():
+	ap = argparse.ArgumentParser()
+	ap.add_argument('-f', '--from', help="folder or xml file", required=True)
+	# ap.add_argument('-t', '--to', help="output folder", default='.')
+	args = ap.parse_args()
+	frompath = args.__dict__['from']
+	return frompath
+
+def get_version():
```
Comment (Yanpas) [inline godotdocset.py]: This code is executed each time, please save to global variable

File: godotdocset.py [hunk 3]
```diff
@@ -131,23 +131,39 @@ def get_plist(name: str):
 	<string>dashtoc</string>
 </dict>
 </plist>
-'''.format(name.split('_')[0],
-		name.replace('_', ' '),
-		name.split('_')[0].lower())
+'''.format(name.split('_')[0]+version, name.replace('_', ' ')+version, name.split('_')[0].lower())
 
 
+def get_args_from():
```
Comment (Yanpas) [inline godotdocset.py]: same here, please return argument parsing to main and save frompath to global variable

File: godotdocset.py [hunk 4]
```diff
@@ -131,23 +131,39 @@ def get_plist(name: str):
...
+		if line.startswith("docs ="):
+			version = "-"+line.split("=")[1].strip().strip('\"')
+	return version
+
 class DocsetMaker:
 	outname = "Godot"
-	rootdir = outname + '.docset'
+	v_outname = outname+get_version()
+	# version_path = os.path.abspath(os.path.join("../", frompath, "./version.py"))
+	rootdir = outname + get_version() + '.docset'
```
Comment (Yanpas) [inline godotdocset.py]: `v_outname + ".docset"`?

**Review by Yanpas — CHANGES_REQUESTED** [review]

File: godotdocset.py (hunk 4 above)
Comment (whytry313) [reply godotdocset.py]: All fixed with new commit, sorry I'm not a python dev, but all these comments are useful to learn

**Pull request merged**